### ISSUE TITLE
Proton games do not launch in KDE Plasma Wayland session 

### ISSUE THREAD

**Issue by ishchow** [flathub/com.valvesoftware.Steam#885]
**Game information**
All games run using Proton. Specifically observed with YS VIII: Lacrimosa of Dana, Nier Replicant v1.22, and Sonic Mania.

**Distribution name and version where applicable**
```
~$ cat /etc/os-release 
NAME="openSUSE Tumbleweed"
# VERSION="20220324"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20220324"
PRETTY_NAME="openSUSE Tumbleweed"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:tumbleweed:20220324"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:Tumbleweed"
LOGO="distributor-logo-Tumbleweed"
```

Desktop environment:
KDE Plasma 5.24.3
Wayland session

**Flatpak info**
```
~$ flatpak --version
Flatpak 1.12.7
```
```
flatpak --gl-drivers
default
host
```

**Problem description**
Steam games running via Proton do not launch at all when Flatpak steam is running in Wayland session. The play button is triggered and then the game process seems to die before drawing any UI.

[Error log](https://gist.github.com/ishchow/c838325227d0fce76c4b42c90e20378f)

Seems there is an earlier issue filed with GNOME Wayland two years ago: #578. But error messages from that issue seem to be different and that issue was fixed by sdk update.

**Does this issue reproduce with native Steam**
Does not reproduce with native steam (either with X11 or Wayland). Does not reproduce with proton games in Flatpak Steam +  X11. Does not reproduce with native games in Flatpak Steam + Wayland (ex. Celeste).

**Comment by Meister1593** [comment]
Same thing
```
[plyshka@fedora plyshka]$ cat /etc/os-release
NAME="Fedora Linux"
VERSION="35.20220411.0 (Kinoite)"
ID=fedora
VERSION_ID=35
VERSION_CODENAME=""
PLATFORM_ID="platform:f35"
PRETTY_NAME="Fedora Linux 35.20220411.0 (Kinoite)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:35"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora-kinoite/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=35
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=35
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Kinoite"
VARIANT_ID=kinoite
OSTREE_VERSION='35.20220411.0'
```
```
[plyshka@fedora plyshka]$ flatpak --version
Flatpak 1.12.7
```
```
[plyshka@fedora plyshka]$ flatpak --gl-drivers
default
host
```
`Plasma 5.24.3`
Running proton through distrobox arch container with steam installed works perfectly fine

**Comment by Meister1593** [comment]
I'm on Gnome Wayland and it works just fine there. But there was update not long ago, maybe it was fixed?

**Comment by sean-gray** [comment]
Im also having this issue. And this is not fixed.

**Comment by junocomp** [comment]
I am also having this issue

**Comment by DaAwesomeP** [comment]
I'm seeing this issue on openSUSE Leap 15.4 with Proton in Steam. I am seeing the `nodrv_CreateWindow` for all games.

**Comment by DaAwesomeP** [comment]
Interestingly, setting Proton to 3.16 seems to work but 7.x, 6.x and Experimental do not work.

**Comment by sean-gray** [comment]
For me on KDE plasma 5.26.9 there is massive flickering. But only on wayland. on x11 its great.

**Comment by DDR0** [comment]
Dang it, I've definitely just reproduced this here, Debian + Wayland yields no window & flickering. If I can provide any details, or help test a fix, let me know.

**Comment by jacobckelly** [comment]
This is still an issue from what I can see. OpenSUSE Aeon \ Gnome \ Wayland

It appears whatever change happened happens between Proton 4.2-9 (which will launch your non-native game) and 4.11-13 (which will just quietly exit).

**Comment by stacyharper** [comment]
From my tests, 5.0-10 works with Celeste, but not 5.13-6, 6.3-8. 7.0-6, 8.0-3, or experimental. But the Flatpak community build 7.0-5 works.

Here Sway on Alpine Linux x86_64, Flatpak Steam, AMD graphical card.

**Comment by stacyharper** [comment]
My issue was that I didn't update the dbus DISPLAY environment variable when xwayland is availabe.

**Comment by doraskayo** [comment]
> My issue was that I didn't update the dbus DISPLAY environment variable when xwayland is availabe.

Can you elaborate on this? Did you run the Steam Flatpak in a separate DBus session?

**Comment by stacyharper** [comment]
> Can you elaborate on this? Did you run the Steam Flatpak in a separate DBus session?

No, I was running dbus before my Sway session, then I was updating the dbus environment variables with `dbus-update-activation-environment WAYLAND_DISPLAY XDG_CURRENT_DESKTOP`. Adding DISPLAY to this list solved my issue.

**Comment by sagehane** [comment]
>Adding DISPLAY to this list solved my issue.

@stacyharper, this solved my issue on River. Kind of obvious in hindsight but you saved me lots of time and confusion, thanks.

**Comment by supagu** [comment]
I am having the same problem with KDE 6 + wayland. I am using EndevourOS.
How would I setup these environment variables for a non-flat pack version of steam to see if that was my issue?
